### PR TITLE
Allow specifying the C++ edition

### PR DESCRIPTION
**Pull request by bwidawsk**
In certain build environments *cough ChromeOS cough* the C++ edition may be hamstrung. For example, chrono may not be available - see https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++-features.md#chrono_banned.

To workaround this, allow an environment variable to specify the edition. For example: `TRACY_CLIENT_CPP_EDITION=gnu++14 cargo build`

Note that since compilers seem to be resilient to badly formed editions, the implementation itself will check for correctness.

### CODE REVIEW

**Comment by Ralith** [comment]
> Note that since compilers seem to be resilient to badly formed editions, the implementation itself will check for correctness.

This seems pointlessly fragile.

File: tracy-client-sys/build.rs [hunk 1]
```diff
@@ -60,12 +60,23 @@ fn set_feature_defines(mut c: cc::Build) -> cc::Build {
...
+        eprintln!("Invalid C++ edition")
+    }
+
     if std::env::var_os("CARGO_FEATURE_ENABLE").is_some() {
         set_feature_defines(cc::Build::new())
             .file("tracy/TracyClient.cpp")
             .warnings(false)
             .cpp(true)
-            .flag_if_supported("-std=c++11")
+            .flag_if_supported(format!("-std={stdcpp}").as_str())
```
[the remark this answers is not in the document]
Comment (nagisa) [reply tracy-client-sys/build.rs:79]: I ended up making a PR to make this super easy to do: https://github.com/rust-lang/cc-rs/pull/818, but in the meanwhile you could also just copy over the relevant pieces of code from `cc-rs` into this build script.

**Review by nagisa — COMMENTED** [review]
I similarly agree that the additional checks for the `-std` values aren’t really necessary. The set of valid values will depend on the version of the compiler, and if anybody wanted to circumvent these checks, they could basically set up a wrapper script over their `$CXX` anyway.

File: tracy-client-sys/build.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ralith) [reply tracy-client-sys/build.rs:79]: Reviewing the linked code, it seems like the PR author's requirements can already be addressed by setting `CXXFLAGS`?

[the remark this answers is not in the document]
Comment (nagisa) [reply tracy-client-sys/build.rs:79]: Unfortunately I don’t believe that’s the case – `CXXFLAGS` will come earlier on the command line, so the `-std=c++11` specified by this crate via `flag_if_supported` will override the `-std` arguments passed in that way.

[the remark this answers is not in the document]
Comment (nagisa) [reply tracy-client-sys/build.rs:79]: Something along the lines of

```rust
let mut builder = set_feature_defines(cc::Build::new());
builder.file(...).warnings(false).cpp(true).flag_if_supported("-std=c++11");
let _ = builder.try_flags_from_environment("TRACY_CLIENT_SYS_CXXFLAGS");
builder....
```

Then whenever `TRACY_CLIENT_SYS_CXXFLAGS=-std=c++14` environment variable is set, the compiler is invoked as

```
cc ... -std=c++11 -std=c++14 $FILE
```

which, yes, has the superfluous `-std=c++11` flag, but the flexibility to set user-specified flags is now available.

[the remark this answers is not in the document]
Comment (bwidawsk) [reply tracy-client-sys/build.rs:79]: Cool thanks! Could I request a version bump for this when it lands in rust_tracy_client?

**Comment by nagisa** [comment]
You can now set the standard you want to use via the `TRACY_CLIENT_SYS_CXXFLAGS` environment variable.

**Pull request closed without merging**